File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/MedicationDispense.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class MedicationDispense : DomainResource, IResource
{
    public Base.ResourceReference? StatusReasonReference { get; set; }
    public Base.CodeableConcept? Category { get; set; }
    public string? WhenHandedOver { get; set; }
    public string? WhenPrepared { get; set; }
    public Base.ResourceReference[]? EventHistory { get; set; }
    public MedicationDispense_Substitution? Substitution { get; set; }
    public Base.ResourceReference[]? DetectedIssue { get; set; }
    public Base.CodeableConcept? MedicationCodeableConcept { get; set; }
    public Base.CodeableConcept? Type { get; set; }
    public object? StatusReason    
    {
        get
        {
            if (StatusReasonReference is not null)
            {
                return StatusReasonReference;
            }
    
            if (StatusReasonCodeableConcept is not null)
            {
                return StatusReasonCodeableConcept;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.ResourceReference))
            {
                StatusReasonReference = (Base.ResourceReference)value;return;
            }
    
            if (value?.GetType() == typeof(Base.CodeableConcept))
            {
                StatusReasonCodeableConcept = (Base.CodeableConcept)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.Annotation[]? Note { get; set; }
    public Base.CodeableConcept? StatusReasonCodeableConcept { get; set; }
    public Base.ResourceReference[]? SupportingInformation { get; set; }
    public required string Status { get; set; }
    public string[]? DosageInstruction { get; set; }
    public Base.Quantity? DaysSupply { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference? Context { get; set; }
    public Base.ResourceReference? MedicationReference { get; set; }
    public Base.Quantity? Quantity { get; set; }
    public Base.ResourceReference[]? PartOf { get; set; }
    public Base.ResourceReference? Location { get; set; }
    public Base.ResourceReference[]? AuthorizingPrescription { get; set; }
    public Base.ResourceReference[]? Receiver { get; set; }
    public Base.ResourceReference? Subject { get; set; }
    public Base.ResourceReference? Destination { get; set; }
    public MedicationDispense_Performer[]? Performer { get; set; }
    public object? Medication    
    {
        get
        {
            if (MedicationCodeableConcept is not null)
            {
                return MedicationCodeableConcept;
            }
    
            if (MedicationReference is not null)
            {
                return MedicationReference;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.CodeableConcept))
            {
                MedicationCodeableConcept = (Base.CodeableConcept)value;return;
            }
    
            if (value?.GetType() == typeof(Base.ResourceReference))
            {
                MedicationReference = (Base.ResourceReference)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }

    public class MedicationDispense_Substitution : BackboneElement
    {
        public Base.CodeableConcept? Type { get; set; }
        public Base.CodeableConcept[]? Reason { get; set; }
        public required bool WasSubstituted { get; set; }
        public Base.ResourceReference[]? ResponsibleParty { get; set; }
    }

    public class MedicationDispense_Performer : BackboneElement
    {
        public required Base.ResourceReference Actor { get; set; }
        public Base.CodeableConcept? Function { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MedicationDispense.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import CodeableConcept
from base import Dosage
from base import Quantity
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class MedicationDispenseSubstitution(BackboneElement):
    type: Optional[CodeableConcept] = None
    reason: Optional[List[CodeableConcept]] = field(default_factory=list)
    was_substituted: bool
    responsible_party: Optional[List[Reference]] = field(default_factory=list)

@dataclass(kw_only=True)
class MedicationDispensePerformer(BackboneElement):
    actor: Reference
    function: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class MedicationDispense(DomainResource):
    status_reason_reference: Optional[Reference] = None
    category: Optional[CodeableConcept] = None
    when_handed_over: Optional[str] = None
    when_prepared: Optional[str] = None
    event_history: Optional[List[Reference]] = field(default_factory=list)
    substitution: Optional[MedicationDispenseSubstitution] = None
    detected_issue: Optional[List[Reference]] = field(default_factory=list)
    medication_codeable_concept: Optional[CodeableConcept] = None
    type: Optional[CodeableConcept] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    status_reason_codeable_concept: Optional[CodeableConcept] = None
    supporting_information: Optional[List[Reference]] = field(default_factory=list)
    status: str
    dosage_instruction: Optional[List[Dosage]] = field(default_factory=list)
    days_supply: Optional[Quantity] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    context: Optional[Reference] = None
    medication_reference: Optional[Reference] = None
    quantity: Optional[Quantity] = None
    part_of: Optional[List[Reference]] = field(default_factory=list)
    location: Optional[Reference] = None
    authorizing_prescription: Optional[List[Reference]] = field(default_factory=list)
    receiver: Optional[List[Reference]] = field(default_factory=list)
    subject: Optional[Reference] = None
    destination: Optional[Reference] = None
    performer: Optional[List[MedicationDispensePerformer]] = field(default_factory=list)